test/tests/performance/pbs_preemptperformance.py

```diff
@@ -480,6 +480,69 @@ def test_preemption_basic(self):
             ncpus *= 3
             S_jobs += ncpus
 
+    @timeout(3600)
+    def test_preemption_with_soft_limits(self):
+        """
+        Measure the time scheduler takes to preempt when the high priority
+        job hits soft limits under a considerable amount of workload.
+        """
+        a = {'resources_available.ncpus': 4,
+             'resources_available.mem': '6400mb'}
+        self.server.create_vnodes('vn', a, 500, self.mom, usenatvnode=False)
+        p = "express_queue, normal_jobs, server_softlimits, queue_softlimits"
+        a = {'preempt_prio': p}
+        self.server.manager(MGR_CMD_SET, SCHED, a, runas=ROOT_USER)
+
+        a = {'max_run_res_soft.ncpus': "[u:" + str(TEST_USER)+"=1]"}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, 'workq')
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+        # submit a bunch of jobs under projectrB
+        a = {ATTR_l + '.select=1:ncpus': 1, ATTR_project: "ProjectB"}
```

Comment:
what's the significance of setting the job's project attribute?

Reply:
that will make all the jobs have the same project and land us in softlimits code

Reply:
Won't all jobs have the same project anyways (_pbs_project_default)? also, the limit is on the user, not the project, so how will setting the project affect limits?